File: 20190917/20190925/830. 较大分组的位置.cc
/*
 * Description: 较大分组的位置
 * Author: AskeyNil
 * CreateDate: 2019-09-25 08:56:00
 * LastEditors: AskeyNil
 *
 * *********************************
 * **                             **
 * **     　  你只有足够努力     　  **
 * **       才能看上去毫不费力       **
 * **                             **
 * *********************************
 *
 */

#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution {
  public:
    vector<vector<int>> largeGroupPositions(string S) {
        vector<vector<int>> vv;
        int leftIndex = 0;
        char left = S[0];
        for (int index = 1; index < S.size(); index++) {
            cout << left << endl;
            if (S[index] != left) {
                if (index - leftIndex >= 3)
                    vv.push_back({leftIndex, index - 1});
                leftIndex = index;
                left = S[index];
            }
        }
        if (S.size() - leftIndex >= 3)
            vv.push_back({leftIndex, int(S.size() - 1)});
        return vv;
    }
};

int main(int argc, char const *argv[]) {
    cout << Solution().largeGroupPositions("aaa")[0][1] << endl;
    return 0;
}

File: 20190917/20190925/830. 较大分组的位置.py
'''
Description: 较大分组的位置
Author: AskeyNil
CreateDate: 2019-09-25 08:55:57
LastEditors: AskeyNil

*********************************
**                             **
**     　  你只有足够努力     　  **
**       才能看上去毫不费力       **
**                             **
*********************************

'''


class Solution:
    def largeGroupPositions(self, S: str) -> [[int]]:
        left = None
        leftIndex = -1
        arr = []
        for index, c in enumerate(S):
            if left == None:
                left = c
                leftIndex = index
            elif c != left:
                if index - leftIndex >= 3:
                    arr.append([leftIndex, index - 1])
                leftIndex = index
                left = c
        if len(S) - leftIndex >= 3:
            arr.append([leftIndex, len(S) - 1])
        return arr
